src/pvq_encoder.c

```diff
@@ -499,6 +501,13 @@ static int pvq_theta(od_coeff *out, const od_coeff *x0, const od_coeff *r0,
       /* Search for the best angle within a reasonable range. */
       qtheta = items[idx].qtheta;
       k = items[idx].k;
+      /* Compute the minimal possible distortion by not taking the PVQ
+         cos_dist into account. */
+      dist_theta = 2 - 2.*od_pvq_cos(theta - qtheta)*OD_TRIG_SCALE_1;
+      dist = gain_weight*(qcg - cg)*(qcg - cg) + qcg*(double)cg*dist_theta;
+      dist *= OD_CGAIN_SCALE_2;
+      /* If we have no hope of beating skip, stop now. */
+      if (dist > dist0 + pvq_norm_lambda && k != 0) continue;
```

Comment:
Instead of doing `&& k != 0` I would put the dist_theta/dist/if inside a `if (k != 0)` and update the comment to mention this is the minimal distortion when k != 0